tg_bot/modules/sql/users_sql.py

```diff
@@ -138,6 +139,20 @@ def get_user_num_chats(user_id):
         SESSION.close()
 
 
+def set_restriction(chat_id, restricted):
+    with INSERTION_LOCK:
+        curr = SESSION.query(Chats).get(str(chat_id))
+        curr.restricted = restricted
```

Comment:
Even though you do a check on all_chats in your code, make sure you have an acceptable failure case here if `curr` is `None`. safety is important!